bothub_nlp_nlu_worker/bothub_nlp_nlu/pipeline_components/preprocessing.py

```diff
@@ -0,0 +1,191 @@
+        # replace regex by TO_WORD
+        phrase = re.sub(to_regex, TO_WORD, phrase)
+        # replace regex by TA_WORD
+        phrase = re.sub(ta_regex, TA_WORD, phrase)
+        # replace regex by MKT_WORD
+        phrase = re.sub(mkt_regex, MKT_WORD, phrase)
+        # replace regex by OK_WORD
+        phrase = re.sub(ok_regex, OK_WORD, phrase)
+        # replace regex by BD_WORD
+        phrase = re.sub(bd_regex, BD_WORD, phrase)
+
+        return phrase
+
+    def english_preprocesing(self, phrase):
+        # set regex for "mkt":
+        mkt_regex = r"\b(mkt)\b"
+        # set regex for "ok":
+        ok_regex = r"\b(ok)\b"
+        # set regex for "ty":
+        ty_regex = r"\b(ty)\b"
+        # set regex for "thx":
+        thx_regex = r"\b(thx)\b"
+        # set regex for "tks":
+        tks_regex = r"\b(tks)\b"
+        # set regex for " 'm / 'mmmm ":
+        am_regex = r"('m)m*\b"
+        # set regex for " 'm / 'mmmm ":
+        are_regex = r"('re)e*\b"
+        # set regex for " *n't ":
+        not_regex = r"(n't)\b"
+
+        # set replace words
+        MKT_WORD = "marketing"
+        OK_WORD = "okay"
+        TY_WORD = "thank you"
+        AM_WORD = " am"
+        ARE_WORD = " are"
+        NOT_WORD = " not"
+
+        # replace regex by MKT_WORD
+        phrase = re.sub(mkt_regex, MKT_WORD, phrase)
+        # replace regex by OK_WORD
+        phrase = re.sub(ok_regex, OK_WORD, phrase)
+        # replace regex by TY_WORD
+        phrase = re.sub(ty_regex, TY_WORD, phrase)
+        # replace regex by THX_WORD
+        phrase = re.sub(thx_regex, TY_WORD, phrase)
+        # replace regex by TKS_WORD
+        phrase = re.sub(tks_regex, TY_WORD, phrase)
+        # replace regex by AM_WORD
+        phrase = re.sub(am_regex, AM_WORD, phrase)
+        # replace regex by ARE_WORD
+        phrase = re.sub(are_regex, ARE_WORD, phrase)
+        # replace regex by NOT_WORD
+        phrase = re.sub(not_regex, NOT_WORD, phrase)
+
+        return phrase
+
+    @classmethod
+    def create(
+        cls, component_config: Dict[Text, Any], config: RasaNLUModelConfig
+    ) -> "Preprocessing":
+        # component_config = override_defaults(cls.defaults, component_config)
+        language = config.language
+        return cls(component_config, language)
+
+    def provide_context(self) -> Dict[Text, Any]:
+        return {'language': self.language}
+
+    def train(
+        self,
+        training_data: TrainingData,
+        config: Optional[RasaNLUModelConfig] = None,
+        **kwargs: Any,
+    ) -> None:
+        """Train this component"""
+        not_repeated_phrases = set()
+        size = len(training_data.training_examples)
+        subtract_idx = 0
+
+        for idx in range(size):
+            example_text = training_data.training_examples[idx - subtract_idx].text
+            # removing accent and lowercasing characters
+            example_text = unidecode(example_text.lower())
+
+            if config.language == "en":
```

Comment:
I don't see this as a good practice, the ideal would be to use the Adapter structural design pattern